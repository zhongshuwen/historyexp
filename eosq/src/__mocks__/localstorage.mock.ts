export class MockStorage implements Storage {
  length = 0

  clearMock = jest.fn()
  getItemMock = jest.fn()
  keyMock = jest.fn()
  removeItemMock = jest.fn()
  setItemMock = jest.fn()

  clear(): void {
    this.clearMock()
  }

  getItem(key: string): string | null {
    return this.getItemMock(key)
  }

  key(index: number): string | null {
    return this.keyMock(index)
  }

  removeItem(key: string): void {
    this.removeItemMock(key)
  }

  setItem(key: string, value: string): void {
    this.setItemMock(key, value)
  }
}
