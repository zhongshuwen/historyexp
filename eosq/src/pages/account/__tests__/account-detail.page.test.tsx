import * as React from "react"
import { shallowWithTheme } from "../../../tests/renderers"
import { getMockRouterProps } from "../../../__mocks__/router"
import { AccountDetail } from "../account-detail.page"

describe("page account-detail", () => {
  it("renders error component if store has error", () => {
    expect(shallowWithTheme(render())).toMatchSnapshot()
  })
})

function render(transactionId: string = "1") {
  const routerProps = getMockRouterProps<{ id: string }>({ id: transactionId })

  return (
    <AccountDetail
      match={routerProps.match}
      location={routerProps.location}
      history={routerProps.history}
    />
  )
}
