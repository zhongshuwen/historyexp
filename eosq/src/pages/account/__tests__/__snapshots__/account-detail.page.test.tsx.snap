// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`page account-detail renders error component if store has error 1`] = `<DataLoading />`;
